lms/static/scripts/frontend_apps/components/test/FilePickerApp-test.js

```diff
@@ -210,8 +210,14 @@ describe('FilePickerApp', () => {
 
       selectContent(wrapper, 'https://example.com');
 
+      const form = wrapper.find('form').getDOMNode();
+      const waitForFormSubmit = new Promise(resolve =>
+        form.addEventListener('submit', resolve)
+      );
+
       // Simulate implicit form submission, as if pressing Enter in title field.
-      wrapper.find('form').getDOMNode().requestSubmit();
+      form.requestSubmit();
+      await waitForFormSubmit;
 
       await waitFor(() => fakeAPICall.called);
       await waitFor(() => onSubmit.called);
```

Comment:
I initially thought we would be able to get rid of these `waitFor` calls, but they seem to still be needed. That makes me have some doubts that my changes actually fixed this, but I think it's worth trying.